packages/@o3r/schematics/src/utility/monorepo.ts

```diff
@@ -25,3 +25,9 @@ export function findConfigFileRelativePath(tree: Tree, files: string[], originPa
 export function isNxContext(tree: Tree) {
   return tree.exists('/nx.json');
 }
+
+/** Default name of the folder where libraries will be generated inside monorepo */
+export const LIBRARIES_FOLDER_NAME = 'libs';
+/** Default name of the folder where applications will be generated inside monorepo */
+export const APPLICATIONS_FOLDER_NAME = 'apps';
+
```

Comment:
would be valuable to provide a function to retrieve the folder name as well.
Something like:

```typescript
export const DEFAULT_ROOT_FOLDERS: WorkspaceLayout = {
  libsDir: LIBRARIES_FOLDER_NAME,
  appsDir: APPLICATIONS_FOLDER_NAME
};
const BASE_ROOT_FOLDERS_MAP: Record<WorkspaceProject['projectType'], keyof WorkspaceLayout> = {
  library: 'libsDir',
  application: 'appsDir'
};

// not in this file :)
export function getSchematicOptions<T extends SchematicOptions['*:*'] = SchematicOptions['*:*']>(config: WorkspaceConfig, schematicName?: string): T | undefined {
  if (!schematicName) {
    return config.schematics?.['*:*'];
  }
  const options = config.schematics && Object.entries(config.schematics)
    .filter(([name]) => new RegExp(name.replace(/\*/g, '.*')).test(schematicName))
    .sort(([a], [b]) => ((a.match(/\*/g)?.length || 0) - (b.match(/\*/g)?.length || 0)))
    .reduce((acc, [, opts]) => ({ ...acc, ...opts }), {} as any);
  return options && Object.keys(options).length ? options : undefined;
}

export function getPackagesBaseRootFolder(config: WorkspaceConfig, tree: Tree, projectType?: WorkspaceProject['projectType'], schematicName?: string) {
  const configName = projectType && BASE_ROOT_FOLDERS_MAP[projectType];
  const nxExplicitDir = configName && isNxContext(tree) && tree.readJson('/nx.json').workspaceLoyout?.[configName];
  const schematicConfigDir = configName && getSchematicConfig(config, schematicName)?.[configName];
  return '/' + (schematicConfigDir || nxExplicitDir || (projectType && DEFAULT_ROOT_FOLDERS[projectType]) || '');
}
```

Reply:
updated